### PR TITLE
make: add and export RIOTTOOLS directory

### PR DESCRIPTION
**Pull request by cladmi**
### Contribution description

Add a `RIOTTOOLS` variable to cleanup build scripts.

This PR will also patch makefiles and scripts to use it.

I started splitting this PR to simplify testing and review.
Also maintaining it is also hard.

* [x] #9067 First PR that justs defines the variable
* [x] #9069 tests: use RIOTTOOLS variable
* [x] #9070 tools/check: use RIOTTOOLS variable
* [x] #9071 sys/arduino: use RIOTTOOLS variable
* [x] #9072 pkg: use RIOTTOOLS variable
* [x] #9102 main Makefile.include related
* [x] #9133 boards
* [x] #9134 flashing
* [x] #9110 examples/gnrc_border_router

The current lines using `dist/tools` on purpose:

```
git grep 'dist/tools' | grep -v -e README.md  -e doc.txt -e '\.md:' -e  static-tests.sh -e 'build_and_test.sh' -e 'changed_files.sh' -e vagrant/bootstrap.sh  -e ':#'
Makefile.include:RIOTTOOLS      ?= $(RIOTBASE)/dist/tools
dist/tools/coccinelle/check.sh:: ${RIOTTOOLS:-./dist/tools}
dist/tools/cppcheck/check.sh:: ${RIOTTOOLS:-./dist/tools}
dist/tools/externc/check.sh:: ${RIOTTOOLS:-./dist/tools}
dist/tools/flake8/check.sh:: ${RIOTTOOLS:-./dist/tools}
dist/tools/headerguards/check.sh:: ${RIOTTOOLS:-./dist/tools}
dist/tools/licenses/check.sh:: ${RIOTTOOLS:-./dist/tools}
dist/tools/pr_check/pr_check.sh:: ${RIOTTOOLS:-./dist/tools}
```

Other usages that I think do not need to be patched
```
git grep -l 'dist/tools' | grep -v -e Makefile.include -e 'check.sh'                                                                                               .drone.yml
.github/ISSUE_TEMPLATE.md
.murdock
README.md
Vagrantfile
cpu/kinetis/dist/check-fcfield-elf.sh
cpu/kinetis/dist/check-fcfield-hex.sh
cpu/native/README.md
dist/tools/ci/build_and_test.sh
dist/tools/ci/changed_files.sh
dist/tools/coccinelle/README.md
dist/tools/cppcheck/README.md
dist/tools/desvirt/README.desvirt.md
dist/tools/git/README.md
dist/tools/packer/README.md
dist/tools/pr_check/check_labels.sh
dist/tools/static-tests.sh
dist/tools/vagrant/README.md
dist/tools/vagrant/bootstrap.sh
doc/doxygen/src/getting-started.md
doc/doxygen/src/mainpage.md
examples/ccn-lite-relay/README.md
examples/dtls-echo/README.md
examples/emcute/README.md
examples/gcoap/README-slip.md
examples/gnrc_border_router/README.md
examples/gnrc_networking/README.md
examples/gnrc_tftp/README.md
makefiles/tests.inc.mk
makefiles/tools/edbg.inc.mk
sys/arduino/doc.txt
```

### Issues/PRs references

First commit was taken from https://github.com/RIOT-OS/RIOT/pull/8509

### CODE REVIEW

**Comment by cladmi** [comment]
The variable is not set when run as its started outside of the `Makefile.include` context but I can add the variable ot the beginning of the script.

File: tests/periph_timer/notes [hunk 1]
```diff
@@ -0,0 +1,14 @@
+arm-none-eabi-readelf  -s bin/mips-malta/periph_common/timer.o | grep timer_set
```
Comment (miri64) [inline tests/periph_timer/notes]: This change looks unrelated.

File: tests/periph_timer/notes_2 [hunk 2]
```diff
@@ -0,0 +1,71 @@
+mips-mti-elf-objdump -d bin/mips-malta/mips32r2_common_periph/timer.o bin/mips-malta/periph_common/timer.o bin/
+mips-malta/*.elf | grep -A 30 '<timer_set>:'
```
Comment (miri64) [inline tests/periph_timer/notes_2]: Dito

File: tests/periph_timer/notes (hunk 1 above)
Comment (cladmi) [reply tests/periph_timer/notes]: Ooops :p

**Comment by cladmi** [comment]
The one in `makefiles/tests.inc.mk` is also not loaded by `Makefile.include` but by the main Makefile.

**Comment by cladmi** [comment]
I updated for the last merged PRs.

File: pkg/ccn-lite/Makefile [hunk 3]
```diff
@@ -11,14 +11,14 @@ TOOLCHAIN_FILE=$(PKG_BUILDDIR)/xcompile-toolchain.cmake
 
 all: $(PKG_BUILDDIR)/src/Makefile
 	$(MAKE) -C $(PKG_BUILDDIR)/src && \
-	cp $(PKG_BUILDDIR)/src/lib/libccnl-riot.a ${BINDIR}/ccn-lite.a
+	cp $(PKG_BUILDDIR)/src/lib/libccnl-riot.a $(BINDIR)/ccn-lite.a
```
Comment (smlng) [inline pkg/ccn-lite/Makefile]: unrelated

File: pkg/ccn-lite/Makefile [hunk 4]
```diff
@@ -11,14 +11,14 @@ TOOLCHAIN_FILE=$(PKG_BUILDDIR)/xcompile-toolchain.cmake
...
 
 $(PKG_BUILDDIR)/src/Makefile: $(TOOLCHAIN_FILE)
 	cd $(PKG_BUILDDIR)/src && \
 	cmake -DCMAKE_TOOLCHAIN_FILE=$(TOOLCHAIN_FILE) \
-		  -DCCNL_RIOT=1 -DRIOT_CFLAGS="${RIOT_CFLAGS}" -DBUILD_TESTING=OFF .
+		  -DCCNL_RIOT=1 -DRIOT_CFLAGS="$(RIOT_CFLAGS)" -DBUILD_TESTING=OFF .
```
Comment (smlng) [inline pkg/ccn-lite/Makefile]: unrelated, too?

File: pkg/ccn-lite/Makefile (hunk 3 above)
Comment (cladmi) [reply pkg/ccn-lite/Makefile]: Yeah I applied https://github.com/RIOT-OS/RIOT/pull/8822 here too, I will move to another PR.

Comment (cladmi) [reply pkg/ccn-lite/Makefile]: Moved to https://github.com/RIOT-OS/RIOT/pull/8907

**Comment by cladmi** [comment]
I started splitting this PR to simplify testing and review.
Also maintaining it is also hard.

First PR that justs defines the variable:

* https://github.com/RIOT-OS/RIOT/pull/9067

**Comment by aabadie** [comment]
@cladmi, I don't think this PR is waiting for other PRs anymore.

**Comment by cladmi** [comment]
I check the last added files and rebase.

**Comment by cladmi** [comment]
I added and successfully re-run the modified tests with native.

**Comment by aabadie** [comment]
Murdock is green, @smlng @miri64 do you ACK ?

**Comment by tcschmidt** [comment]
Ping @miri64 @smlng

**Comment by smlng** [comment]
@cladmi: if you deem this ready, please hit merge ...

**Comment by cladmi** [comment]
I rebased and re-reviewed the uses of `dist/tools` in the repository. Everything seems still up to date. Merging when murdock is green.
